lib/ecto_shorts/config.ex

```diff
@@ -3,7 +3,94 @@ defmodule EctoShorts.Config do
 
   @app :ecto_shorts
 
+  @doc """
+  Returns the value of `ecto_shorts` config key `:repo`.
+
+  ### Examples
+
+      iex> EctoShorts.Config.repo()
+      EctoShorts.Support.Repo
+  """
+  @spec repo :: Ecto.Repo.t() | nil
   def repo do
     Application.get_env(@app, :repo)
   end
+
+  @doc """
+  Returns the value of `ecto_shorts` config key `:replica`.
+
+  ### Examples
+
+      iex> EctoShorts.Config.replica()
+      nil
+  """
+  @doc since: "2.5.0"
+  @spec replica :: Ecto.Repo.t() | nil
+  def replica do
+    Application.get_env(@app, :replica)
+  end
+
+  @doc """
+  Returns a `Ecto.Repo` module.
+
+  Raises if the repo is not configured and the option `:repo` is not set.
+
+  ### Examples
+
+      iex> EctoShorts.Config.repo!()
+      EctoShorts.Support.Repo
+
+      iex> EctoShorts.Config.repo!(repo: YourApp.Repo)
+      YourApp.Repo
+  """
+  @doc since: "2.5.0"
+  @spec repo!(keyword()) :: Ecto.Repo.t()
+  @spec repo! :: Ecto.Repo.t()
+  def repo!(opts \\ []) do
+    with nil <- Keyword.get(opts, :repo, repo()) do
+      raise ArgumentError, """
+      EctoShorts must be configured with a repo.
+
+      To fix this error you can do one of the following:
+
+      1. Configure the repo:
+
+      ```
+      config :ecto_shorts, :repo, YourApp.Repo
+      ```
+
+      2. Pass in the `:repo` option:
+
+      ```
+      [repo: YourApp.Repo]
+      ```
+      """
+    end
+  end
+
+  @doc """
+  Returns a `Ecto.Repo` module.
+
+  This function attempts to retrieve a repo from the `:replica` option
+  and will fallback to returning the value from the `:repo` option or
+  the configured repo if the option `:replica` is not set.
+
+  Raises if no repo is found.
+
+  ### Examples
+
+      iex> EctoShorts.Config.replica!()
+      EctoShorts.Support.Repo
+
+      iex> EctoShorts.Config.replica!(replica: YourApp.Repo.Replica)
+      YourApp.Repo.Replica
+  """
+  @doc since: "2.5.0"
+  @spec replica!(keyword()) :: Ecto.Repo.t()
+  @spec replica! :: Ecto.Repo.t()
+  def replica!(opts \\ []) do
+    with nil <- Keyword.get(opts, :replica, replica()) do
+      repo!(opts)
```

Comment:
This part is confusing because the error will complain about `:repo` when it's actually needing to set `:replica` might need a different error here

Reply:
@MikaAK updated